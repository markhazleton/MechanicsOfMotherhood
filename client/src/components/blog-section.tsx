import { Wrench } from "lucide-react";
import { useLocation } from "wouter";
import { Button } from "@/components/ui/button";

export default function BlogSection() {
  const [, navigate] = useLocation();
  
  return (
    <section id="blog" className="py-16 bg-kitchen-warm">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
        <div className="text-center mb-12">
          <div className="flex items-center justify-center mb-4">
            <Wrench className="text-tool-gray text-2xl mr-3" />
            <h2 className="font-industrial text-3xl md:text-4xl font-bold text-industrial-blue">
              MoM's Maintenance Manual
            </h2>
            <Wrench className="text-tool-gray text-2xl ml-3" />
          </div>
          <p className="text-tool-gray text-lg max-w-2xl mx-auto">
            Real stories, practical tips, and maintenance logs from the trenches of working motherhood.
          </p>
        </div>

        {/* Coming Soon Message */}
        <div className="bg-white rounded-xl p-8 mechanical-shadow text-center max-w-md mx-auto">
          <div className="w-16 h-16 bg-workshop-teal rounded-full flex items-center justify-center mx-auto mb-4">
            <Wrench className="w-8 h-8 text-white" />
          </div>
          <h3 className="font-mechanical text-xl font-bold text-industrial-blue mb-3">
            Manual Coming Soon!
          </h3>
          <p className="text-tool-gray mb-6">
            We're crafting insightful articles and practical tips for working mothers. 
            Check back soon for our maintenance manual!
          </p>
        </div>

        <div className="text-center mt-12">
          <Button
            variant="outline"
            size="lg"
            className="gear-border bg-white hover:bg-gray-50 text-tool-gray px-8 py-4 rounded-xl font-semibold text-lg transition-all"
            data-testid="button-view-complete-manual"
            onClick={() => navigate('/blog')}
          >
            <Wrench className="mr-2" size={20} />
            View Blog Page
          </Button>
        </div>
      </div>
    </section>
  );
}
