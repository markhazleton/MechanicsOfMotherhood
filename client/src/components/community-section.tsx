import { useState } from "react";
import { useQuery } from "@tanstack/react-query";
import { Users, MessageCircle, Lightbulb, Rocket } from "lucide-react";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";
import { Card, CardContent } from "@/components/ui/card";
import { useToast } from "@/hooks/use-toast";
import LoadingSpinner from "./loading-spinner";
import apiData from "@/data/api-data.json";

export default function CommunitySection() {
  const [email, setEmail] = useState("");
  const { toast } = useToast();

  const { metadata: stats } = apiData;
  const isLoading = !stats;

  const handleSubmit = (e: React.FormEvent) => {
    e.preventDefault();
    if (email && email.includes("@")) {
      // Static app - show success message without actual submission
      toast({
        title: "Thanks for your interest!",
        description: "We're working on newsletter functionality. Please check back soon!",
      });
      setEmail("");
    } else {
      toast({
        title: "Invalid Email",
        description: "Please enter a valid email address.",
        variant: "destructive",
      });
    }
  };

  if (isLoading) {
    return (
      <section className="py-16 bg-white">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          <LoadingSpinner />
        </div>
      </section>
    );
  }

  return (
    <section className="py-16 bg-white">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
        <div className="text-center mb-12">
          <h2 className="font-industrial text-3xl md:text-4xl font-bold text-industrial-blue mb-4">
            Join the MoM Crew
          </h2>
          <p className="text-tool-gray text-lg max-w-2xl mx-auto">
            Connect with fellow working moms who understand the daily grind. Share tips, troubleshoot problems, and celebrate wins together.
          </p>
        </div>

        <div className="grid grid-cols-1 md:grid-cols-3 gap-8 mb-12">
          <Card className="text-center p-6 bg-kitchen-warm rounded-xl border-none">
            <CardContent className="pt-6">
              <div className="bg-energetic-orange w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4">
                <Users className="text-white text-2xl" />
              </div>
              <h3 className="font-bold text-xl mb-2 text-industrial-blue">Active Community</h3>
              <p className="text-tool-gray mb-4">
                Join <span className="font-semibold" data-testid="community-members-count">
                  {stats?.totalRecipes ? (stats.totalRecipes * 100).toLocaleString() : "15,000"}+
                </span> working moms sharing real solutions.
              </p>
            </CardContent>
          </Card>
          
          <Card className="text-center p-6 bg-kitchen-warm rounded-xl border-none">
            <CardContent className="pt-6">
              <div className="bg-workshop-teal w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4">
                <MessageCircle className="text-white text-2xl" />
              </div>
              <h3 className="font-bold text-xl mb-2 text-industrial-blue">Daily Support</h3>
              <p className="text-tool-gray mb-4">
                Get answers to your questions from experienced mom engineers.
              </p>
            </CardContent>
          </Card>
          
          <Card className="text-center p-6 bg-kitchen-warm rounded-xl border-none">
            <CardContent className="pt-6">
              <div className="bg-yellow-500 w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4">
                <Lightbulb className="text-white text-2xl" />
              </div>
              <h3 className="font-bold text-xl mb-2 text-industrial-blue">Exclusive Content</h3>
              <p className="text-tool-gray mb-4">
                Access member-only recipes, meal plans, and workshop sessions.
              </p>
            </CardContent>
          </Card>
        </div>

        <div className="bg-gradient-to-r from-industrial-blue to-workshop-teal rounded-2xl p-8 md:p-12 text-center text-white">
          <h3 className="font-mechanical text-2xl md:text-3xl font-bold mb-4">
            Ready to Join the Workshop?
          </h3>
          <p className="text-xl mb-8 opacity-90">
            Get instant access to our complete recipe database and join the most supportive community of working moms.
          </p>
          <form onSubmit={handleSubmit} className="flex flex-col sm:flex-row gap-4 justify-center items-center max-w-md mx-auto">
            <Input
              type="email"
              placeholder="Enter your email address"
              value={email}
              onChange={(e) => setEmail(e.target.value)}
              className="flex-1 px-6 py-4 rounded-xl text-tool-gray focus:ring-2 focus:ring-energetic-orange focus:outline-none"
              data-testid="newsletter-email-input"
              required
            />
            <Button
              type="submit"
              className="bg-energetic-orange hover:bg-red-600 text-white px-8 py-4 rounded-xl font-semibold whitespace-nowrap transition-all"
              data-testid="button-start-building"
            >
              <Rocket className="mr-2" size={18} />
              Start Building
            </Button>
          </form>
          <p className="text-sm opacity-75 mt-4">
            Join for free. No spam, just practical tips and recipes.
          </p>
        </div>
      </div>
    </section>
  );
}
