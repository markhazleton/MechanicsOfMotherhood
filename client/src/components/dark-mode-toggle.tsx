import { useEffect, useState } from 'react';
import { Moon, Sun } from 'lucide-react';
import { Button } from '@/components/ui/button';

// Simple dark mode toggle using class strategy (tailwind.config.ts darkMode: 'class')
// Persists preference in localStorage under key 'mom-theme'
export function DarkModeToggle({ className = '' }: { className?: string }) {
  const [isDark, setIsDark] = useState(false);

  useEffect(() => {
    const stored = typeof window !== 'undefined' ? localStorage.getItem('mom-theme') : null;
    const prefersDark = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
    const enabled = stored ? stored === 'dark' : prefersDark;
    setIsDark(enabled);
    document.documentElement.classList.toggle('dark', enabled);
  }, []);

  const toggle = () => {
    setIsDark(prev => {
      const next = !prev;
      document.documentElement.classList.toggle('dark', next);
      try {
        localStorage.setItem('mom-theme', next ? 'dark' : 'light');
      } catch {}
      return next;
    });
  };

  return (
    <Button
      type="button"
      variant="outlineBrand"
      size="sm"
      aria-label={isDark ? 'Activate light mode' : 'Activate dark mode'}
      onClick={toggle}
      className={className}
    >
      {isDark ? (
        <Sun className="size-4" />
      ) : (
        <Moon className="size-4" />
      )}
      <span className="sr-only">{isDark ? 'Light Mode' : 'Dark Mode'}</span>
    </Button>
  );
}

export default DarkModeToggle;
