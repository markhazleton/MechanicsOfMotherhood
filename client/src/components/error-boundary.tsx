import React, { Component, ReactNode } from 'react';
import { Button } from '@/components/ui/button';

interface Props {
  children: ReactNode;
  fallback?: ReactNode;
  onErrorCapture?: (error: Error, info: React.ErrorInfo, errorId: string) => void;
}

interface State {
  hasError: boolean;
  error?: Error;
  transient?: boolean;
}

export class ErrorBoundary extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { hasError: false };
  }

  static getDerivedStateFromError(error: Error): State {
    return { hasError: true, error };
  }

  componentDidCatch(error: Error, errorInfo: React.ErrorInfo) {
    const isDomNotFound = (error as any).name === 'NotFoundError';
    // Suppress noisy React 19 DOM race errors (removeChild on detached node) that are non-fatal
    if (!isDomNotFound) {
      console.error('Error caught by boundary:', error, errorInfo);
    } else if (!import.meta.env.PROD) {
      // Log once in development for awareness
      console.warn('[suppressed] NotFoundError (likely hydration reconciliation race)');
    }
    const errorId = Date.now().toString(36) + '-' + Math.random().toString(36).slice(2,8);
    if (this.props.onErrorCapture) {
      try { this.props.onErrorCapture(error, errorInfo, errorId); } catch (e) {
        console.warn('onErrorCapture handler failed', e);
      }
    }
    if (import.meta.env.PROD) {
      // Placeholder for production logging integration
      // navigator.sendBeacon('/__log', JSON.stringify({ error: error.message, stack: error.stack, errorId }));
    }
    // For NotFoundError we avoid setting error state to prevent UI swap & loop
    if (isDomNotFound) return;
  }

  handleReset = () => {
    this.setState({ hasError: false, error: undefined });
  };

  render() {
    if (this.state.hasError) {
      if (this.state.transient) {
        return (
          <div className="min-h-screen flex items-center justify-center text-sm text-gray-500">
            <div>Recoveringâ€¦</div>
          </div>
        );
      }
      if (this.props.fallback) return this.props.fallback;

      return (
        <div className="min-h-screen bg-gradient-to-b from-slate-50 to-white flex items-center justify-center">
          <div className="max-w-md mx-auto text-center p-6">
            <div className="mb-6">
              <div className="mx-auto w-16 h-16 bg-red-100 rounded-full flex items-center justify-center mb-4">
                <svg className="w-8 h-8 text-red-500" fill="none" stroke="currentColor" viewBox="0 0 24 24">
                  <path strokeLinecap="round" strokeLinejoin="round" strokeWidth={2} d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-2.5L13.732 4c-.77-.833-1.964-.833-2.732 0L3.732 16.5c-.77.833.192 2.5 1.732 2.5z" />
                </svg>
              </div>
              
              <h1 className="text-2xl font-bold text-gray-900 mb-2">
                Something went wrong
              </h1>
              
              <p className="text-gray-600 mb-4">
                We're sorry, but something unexpected happened. Please try refreshing the page.
              </p>
              <p className="text-xs text-gray-400 mb-6">If the issue persists contact support and share this code: <code>{this.state.error?.name}</code></p>
              
              {!import.meta.env.PROD && this.state.error && (
                <details className="text-left bg-gray-100 p-4 rounded-lg mb-6">
                  <summary className="cursor-pointer font-medium text-gray-700">
                    Error Details (Development)
                  </summary>
                  <pre className="mt-2 text-sm text-red-600 whitespace-pre-wrap">
                    {this.state.error.stack}
                  </pre>
                </details>
              )}
            </div>
            
            <div className="space-y-3">
              <Button 
                onClick={this.handleReset}
                className="w-full"
              >
                Try Again
              </Button>
              
              <Button 
                variant="outline" 
                onClick={() => window.location.href = '/'}
                className="w-full"
              >
                Go Home
              </Button>
              
              <Button 
                variant="ghost" 
                onClick={() => window.location.reload()}
                className="w-full"
              >
                Refresh Page
              </Button>
            </div>
          </div>
        </div>
      );
    }

    return this.props.children;
  }
}

// HOC for wrapping components with error boundary
export function withErrorBoundary<P extends object>(
  Component: React.ComponentType<P>,
  fallback?: ReactNode
) {
  const WrappedComponent = (props: P) => (
    <ErrorBoundary fallback={fallback}>
      <Component {...props} />
    </ErrorBoundary>
  );
  
  WrappedComponent.displayName = `withErrorBoundary(${Component.displayName || Component.name})`;
  
  return WrappedComponent;
}
