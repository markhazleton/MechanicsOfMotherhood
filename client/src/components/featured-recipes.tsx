import { useQuery } from "@tanstack/react-query";
import { Clock, Users, TrendingUp, Star, ArrowRight, Settings } from "lucide-react";
import { Button } from "@/components/ui/button";
import { Card, CardContent } from "@/components/ui/card";
import { Badge } from "@/components/ui/badge";
import LoadingSpinner from "./loading-spinner";
import type { Recipe, Category } from "@shared/schema";

export default function FeaturedRecipes() {
  const { data: featuredContent, isLoading } = useQuery({
    queryKey: ["/api/featured-content"],
  });

  const { data: categoriesData } = useQuery({
    queryKey: ["/api/categories"],
  });

  const categories = categoriesData?.categories || [];
  const recipes = featuredContent?.recipes || [];

  const getDifficultyIcon = (difficulty: string) => {
    switch (difficulty) {
      case "easy": return <TrendingUp size={14} />;
      case "medium": return <TrendingUp size={14} />;
      case "hard": return <TrendingUp size={14} />;
      default: return <TrendingUp size={14} />;
    }
  };

  const getCategoryColor = (categoryName: string) => {
    const category = categories.find((c: Category) => c.name === categoryName);
    return category?.color || "#38B2AC";
  };

  if (isLoading) {
    return (
      <section className="py-16 bg-white">
        <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
          <LoadingSpinner />
        </div>
      </section>
    );
  }

  return (
    <section id="recipes" className="py-16 bg-white">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
        <div className="text-center mb-12">
          <div className="flex items-center justify-center mb-4">
            <Settings className="text-tool-gray text-2xl mr-3" />
            <h2 className="font-industrial text-3xl md:text-4xl font-bold text-industrial-blue">
              MoM's Kitchen Workshop
            </h2>
            <Settings className="text-tool-gray text-2xl ml-3" />
          </div>
          <p className="text-tool-gray text-lg max-w-2xl mx-auto">
            Precision-engineered recipes for the modern working mother. Every dish tested, timed, and perfected.
          </p>
        </div>

        {/* Category Filters */}
        <div className="flex flex-wrap justify-center gap-4 mb-12">
          {categories.map((category: Category) => (
            <Button
              key={category.id}
              variant="outline"
              className="bg-kitchen-warm hover:bg-energetic-orange hover:text-white text-tool-gray px-6 py-3 rounded-full font-medium transition-all duration-300 border border-medium-gray"
              style={{ '--hover-bg': category.color } as React.CSSProperties}
              data-testid={`category-button-${category.name.toLowerCase().replace(/\s+/g, '-')}`}
            >
              <Clock size={16} className="mr-2" />
              {category.name} ({category.recipeCount})
            </Button>
          ))}
        </div>

        {/* Recipe Grid */}
        <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-8">
          {recipes.map((recipe: Recipe) => (
            <Card key={recipe.id} className="gear-border bg-white rounded-xl overflow-hidden mechanical-shadow hover:transform hover:scale-105 transition-all duration-300" data-testid={`recipe-card-${recipe.id}`}>
              <img
                src={recipe.imageUrl}
                alt={recipe.title}
                className="w-full h-48 object-cover"
              />
              <CardContent className="p-6">
                <div className="flex items-center justify-between mb-3">
                  <Badge
                    className="text-white px-3 py-1 rounded-full text-sm font-medium"
                    style={{ backgroundColor: getCategoryColor(recipe.category) }}
                  >
                    <Clock size={14} className="mr-1" />
                    {recipe.prepTime + recipe.cookTime} MIN
                  </Badge>
                  <div className="flex items-center text-energetic-orange">
                    <Star size={14} fill="currentColor" />
                    <span className="ml-1 text-sm font-semibold">{recipe.rating}</span>
                  </div>
                </div>
                <h3 className="font-bold text-xl mb-2 text-industrial-blue" data-testid={`recipe-title-${recipe.id}`}>
                  {recipe.title}
                </h3>
                <p className="text-tool-gray mb-4">
                  {recipe.description}
                </p>
                <div className="flex items-center justify-between">
                  <div className="flex items-center space-x-4 text-sm text-tool-gray">
                    <span className="flex items-center">
                      <Users size={14} className="mr-1" />
                      {recipe.servings} servings
                    </span>
                    <span className="flex items-center">
                      {getDifficultyIcon(recipe.difficulty)}
                      <span className="ml-1 capitalize">{recipe.difficulty}</span>
                    </span>
                  </div>
                  <Button
                    variant="ghost"
                    size="sm"
                    className="text-energetic-orange hover:text-red-600 font-semibold"
                    data-testid={`button-build-recipe-${recipe.id}`}
                  >
                    <ArrowRight size={16} className="mr-1" />
                    Build It
                  </Button>
                </div>
              </CardContent>
            </Card>
          ))}
        </div>

        <div className="text-center mt-12">
          <Button
            variant="outline"
            size="lg"
            className="gear-border bg-white hover:bg-gray-50 text-tool-gray px-8 py-4 rounded-xl font-semibold text-lg transition-all"
            data-testid="button-browse-workshop-manual"
          >
            <Settings className="mr-2" size={20} />
            Browse Full Workshop Manual
          </Button>
        </div>
      </div>
    </section>
  );
}
