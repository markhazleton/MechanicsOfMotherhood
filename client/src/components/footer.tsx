import { Settings, Utensils, Facebook, Instagram, Youtube } from "lucide-react";
import { Link } from "wouter";

export default function Footer() {
  const currentYear = new Date().getFullYear();

  return (
    <footer className="bg-industrial-blue text-white py-16">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8">
        <div className="grid grid-cols-1 md:grid-cols-4 gap-8 mb-8">
          {/* Brand Column */}
          <div className="md:col-span-1">
            <div className="flex items-center space-x-3 mb-4">
              <div className="relative">
                <img 
                  src="/images/logos/MOM-Logo-Icon.png"
                  alt="MoM Logo Icon"
                  className="h-8 w-8 object-contain filter brightness-0 invert"
                />
              </div>
              <div>
                <h3 className="font-mechanical text-xl font-bold">MoM</h3>
                <p className="text-xs text-gray-300 font-industrial">MECHANICS OF MOTHERHOOD</p>
              </div>
            </div>
            <p className="text-gray-300 mb-4">
              Engineering better meals for working mothers worldwide.
            </p>
            <div className="flex space-x-4">
              <a
                href="#"
                className="text-gray-300 hover:text-energetic-orange transition-colors"
                aria-label="Facebook"
                data-testid="social-link-facebook"
              >
                <Facebook size={20} />
              </a>
              <a
                href="#"
                className="text-gray-300 hover:text-energetic-orange transition-colors"
                aria-label="Instagram"
                data-testid="social-link-instagram"
              >
                <Instagram size={20} />
              </a>
              <a
                href="#"
                className="text-gray-300 hover:text-energetic-orange transition-colors"
                aria-label="YouTube"
                data-testid="social-link-youtube"
              >
                <Youtube size={20} />
              </a>
            </div>
          </div>

          {/* Workshop Links */}
          <div>
            <h4 className="font-industrial text-lg font-semibold mb-4">Workshop</h4>
            <ul className="space-y-2">
              <li>
                <Link href="/recipes" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-recipe-manual">
                  Recipe Manual
                </Link>
              </li>
              <li>
                <Link href="/recipes?category=quick-fixes" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-quick-fixes">
                  Quick Fixes
                </Link>
              </li>
              <li>
                <Link href="/recipes?category=kid-friendly" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-kid-friendly">
                  Kid-Friendly Builds
                </Link>
              </li>
              <li>
                <Link href="/recipes?category=meal-prep" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-meal-prep">
                  Meal Prep Systems
                </Link>
              </li>
              <li>
                <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-kitchen-tools">
                  Kitchen Tools
                </a>
              </li>
            </ul>
          </div>

          {/* Resources */}
          <div>
            <h4 className="font-industrial text-lg font-semibold mb-4">Resources</h4>
            <ul className="space-y-2">
              <li>
                <Link href="/blog" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-maintenance-manual">
                  Maintenance Manual
                </Link>
              </li>
              <li>
                <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-troubleshooting">
                  Troubleshooting Guide
                </a>
              </li>
              <li>
                <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-community-forum">
                  Community Forum
                </a>
              </li>
              <li>
                <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-video-tutorials">
                  Video Tutorials
                </a>
              </li>
              <li>
                <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-printable-plans">
                  Printable Plans
                </a>
              </li>
            </ul>
          </div>

          {/* Company */}
          <div>
            <h4 className="font-industrial text-lg font-semibold mb-4">Company</h4>
            <ul className="space-y-2">
              <li>
                <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-about-mom">
                  About MoM
                </a>
              </li>
              <li>
                <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-our-story">
                  Our Story
                </a>
              </li>
              <li>
                <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-contact">
                  Contact Us
                </a>
              </li>
              <li>
                <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-partnerships">
                  Partnerships
                </a>
              </li>
              <li>
                <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-press-kit">
                  Press Kit
                </a>
              </li>
            </ul>
          </div>
        </div>

        {/* Bottom Bar */}
        <div className="border-t border-gray-700 pt-8 flex flex-col md:flex-row justify-between items-center">
          <p className="text-gray-300 text-sm mb-4 md:mb-0" data-testid="copyright-text">
            Â© {currentYear} Mechanics of Motherhood. All rights reserved.
          </p>
          <div className="flex space-x-6 text-sm">
            <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-privacy">
              Privacy Policy
            </a>
            <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-terms">
              Terms of Service
            </a>
            <a href="#" className="text-gray-300 hover:text-white transition-colors" data-testid="footer-link-cookies">
              Cookie Policy
            </a>
          </div>
        </div>
      </div>
    </footer>
  );
}
