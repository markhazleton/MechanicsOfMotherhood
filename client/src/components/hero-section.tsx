import { useQuery } from "@tanstack/react-query";
import { Settings, Utensils, Play, Wrench } from "lucide-react";
import { Button } from "@/components/ui/button";
import LoadingSpinner from "./loading-spinner";
import logoIcon from "@/assets/MOM-Logo-Icon.png";
import type { StatsResponse } from "@/data/api-types";

export default function HeroSection() {
  const { data: stats, isLoading } = useQuery<StatsResponse>({
    queryKey: ["/api/stats"],
  });

  return (
    <section className="bg-gradient-to-br from-industrial-blue via-tool-gray to-workshop-teal py-20">
      <div className="max-w-7xl mx-auto px-4 sm:px-6 lg:px-8 text-center">
        <div className="flex justify-center items-center mb-8">
          <div className="relative">
            <div className="bg-white rounded-full p-6 mechanical-shadow">
              <img 
                src={logoIcon}
                alt="MoM Logo"
                className="h-16 w-16 object-contain"
              />
            </div>
          </div>
        </div>
        
        <h1 className="font-mechanical text-5xl md:text-7xl font-black text-white mb-4" data-testid="hero-title">
          MoM
        </h1>
        <p className="font-industrial text-xl md:text-2xl text-gray-200 mb-2 tracking-wide">
          MECHANICS OF MOTHERHOOD
        </p>
        <p className="text-lg md:text-xl text-gray-300 mb-8 font-medium">
          Engineering Better Meals â€¢ The Mother of All Solutions
        </p>
        
        <div className="flex flex-col sm:flex-row gap-4 justify-center items-center mb-12">
          <Button 
            size="lg"
            className="gear-border bg-energetic-orange hover:bg-red-600 text-white px-8 py-4 rounded-xl font-semibold text-lg transition-all duration-300 transform hover:scale-105"
            data-testid="button-start-cooking"
          >
            <Wrench className="mr-2" size={20} />
            Start Cooking
          </Button>
          <Button
            variant="outline"
            size="lg"
            className="bg-white/10 backdrop-blur-sm border border-white/20 text-white px-8 py-4 rounded-xl font-semibold text-lg hover:bg-white/20 transition-all"
            data-testid="button-watch-tutorial"
          >
            <Play className="mr-2" size={20} />
            Watch Tutorial
          </Button>
        </div>

        {/* Stats */}
        <div className="grid grid-cols-1 md:grid-cols-3 gap-8 max-w-2xl mx-auto">
          {isLoading ? (
            <div className="md:col-span-3">
              <LoadingSpinner />
            </div>
          ) : stats ? (
            <>
              <div className="text-center" data-testid="stat-recipes">
                <div className="text-3xl font-bold text-white">{stats.recipes}+</div>
                <div className="text-gray-300 font-industrial">Tested Recipes</div>
              </div>
              <div className="text-center" data-testid="stat-families">
                <div className="text-3xl font-bold text-white">{(stats.families || 25000).toLocaleString()}+</div>
                <div className="text-gray-300 font-industrial">Families Served</div>
              </div>
              <div className="text-center" data-testid="stat-time-saved">
                <div className="text-3xl font-bold text-white">{((stats.timeSaved || 5000000) / 1000000).toFixed(1)}M+</div>
                <div className="text-gray-300 font-industrial">Hours Saved</div>
              </div>
            </>
          ) : (
            <div className="md:col-span-3 text-white">Unable to load statistics</div>
          )}
        </div>
      </div>
    </section>
  );
}
