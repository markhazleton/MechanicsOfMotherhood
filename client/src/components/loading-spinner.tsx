import { Settings, Utensils } from "lucide-react";

export default function LoadingSpinner() {
  return (
    <div className="flex items-center justify-center p-8">
      <div className="relative">
    <Settings className="text-brand-tool text-4xl animate-spin-slow opacity-20" />
        <div className="absolute inset-0 flex items-center justify-center">
          <div className="bg-white rounded-full p-2 mechanical-shadow">
      <Utensils className="text-brand-orange text-lg animate-pulse" />
          </div>
        </div>
      </div>
    </div>
  );
}
