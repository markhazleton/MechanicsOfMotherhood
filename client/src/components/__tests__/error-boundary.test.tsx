import { describe, it, expect, beforeAll, afterAll, vi } from 'vitest';
import { render } from '@testing-library/react';
import React from 'react';
import { ErrorBoundary } from '@/components/error-boundary';

function ProblemChild() {
  throw new Error('Boom');
  return null;
}

describe('ErrorBoundary', () => {
  let originalError: typeof console.error;
  beforeAll(() => {
    // Silence expected React error logging for the intentional test crash
    originalError = console.error;
    vi.spyOn(console, 'error').mockImplementation(() => {});
  });
  afterAll(() => {
    console.error = originalError;
  });

  it('renders fallback UI when child throws', () => {
    const { getByText, container } = render(
      <ErrorBoundary>
        <ProblemChild />
      </ErrorBoundary>
    );
    expect(getByText(/Something went wrong/i)).toBeInTheDocument();
    // Snapshot (strip dynamic pieces like stack traces / error codes)
    const html = container.innerHTML
      .replace(/code>.*?<\/code>/g, 'code>ErrorName</code>')
      .replace(/Error Details \(Development\)[\s\S]*/g, '');
    expect(html).toMatchSnapshot();
  });

  it('renders children when no error', () => {
    const { getByText } = render(
      <ErrorBoundary>
        <div>Safe Child</div>
      </ErrorBoundary>
    );
    expect(getByText('Safe Child')).toBeInTheDocument();
  });
});
