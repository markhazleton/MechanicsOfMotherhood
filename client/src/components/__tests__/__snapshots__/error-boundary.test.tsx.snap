// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorBoundary > renders fallback UI when child throws 1`] = `"<div class="min-h-screen bg-gradient-to-b from-slate-50 to-white flex items-center justify-center"><div class="max-w-md mx-auto text-center p-6"><div class="mb-6"><div class="mx-auto w-16 h-16 bg-red-100 rounded-full flex items-center justify-center mb-4"><svg class="w-8 h-8 text-red-500" fill="none" stroke="currentColor" viewBox="0 0 24 24"><path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M12 9v2m0 4h.01m-6.938 4h13.856c1.54 0 2.502-1.667 1.732-2.5L13.732 4c-.77-.833-1.964-.833-2.732 0L3.732 16.5c-.77.833.192 2.5 1.732 2.5z"></path></svg></div><h1 class="text-2xl font-bold text-gray-900 mb-2">Something went wrong</h1><p class="text-gray-600 mb-4">We're sorry, but something unexpected happened. Please try refreshing the page.</p><p class="text-xs text-gray-400 mb-6">If the issue persists contact support and share this code: <code>ErrorName</code></p><details class="text-left bg-gray-100 p-4 rounded-lg mb-6"><summary class="cursor-pointer font-medium text-gray-700">"`;
